eng/SourceBuild.props

```diff
@@ -0,0 +1,69 @@
+<Project>
+
+  <PropertyGroup>
+    <GitHubRepositoryName>aspnetcore</GitHubRepositoryName>
+    <SourceBuildManagedOnly>true</SourceBuildManagedOnly>
+    <CloneSubmodulesToInnerSourceBuildRepo>false</CloneSubmodulesToInnerSourceBuildRepo>
+  </PropertyGroup>
+
+  <Target Name="ApplySourceBuildPatchFiles"
+          AfterTargets="PrepareInnerSourceBuildRepoRoot"
+          BeforeTargets="RunInnerSourceBuildCommand">
+
+    <ItemGroup>
+      <SourceBuildPatchFile Include="$(RepositoryEngineeringDir)source-build-patches\*.patch" />
+    </ItemGroup>
+
+    <Exec
+      Command="git apply --ignore-whitespace --whitespace=nowarn &quot;%(SourceBuildPatchFile.FullPath)&quot;"
+      WorkingDirectory="$(InnerSourceBuildRepoRoot)"
+      Condition="'@(SourceBuildPatchFile)' != ''" />
+  </Target>
+
+  <!--
+    Init submodules - temporarary workaround for https://github.com/dotnet/sourcelink/pull/653
+  -->
+  <Target Name="InitSubmodules"
+          DependsOnTargets="PrepareInnerSourceBuildRepoRoot"
+          BeforeTargets="RunInnerSourceBuildCommand">
+
+    <Exec
+      Command="git submodule update --init --recursive"
+      WorkingDirectory="$(InnerSourceBuildRepoRoot)"
+      EnvironmentVariables="@(InnerBuildEnv)" />
+  </Target>
+
+  <!--
+    Remove inner source .globalconfig file as both the inner and outer config files get loaded and cause a conflict.
+    Leaving the inner will cause all conflicting settings to be ignored.
+    https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files#general-options
```

Comment:
What's the TLDR info we need to understand "inner and outer" in this context as well as w/ respect to "inner builds"❔ I _think_ inner builds will happen in one or more Docker containers after some setup on the "outer" build agent but am not sure&hellip;

Reply:
The Arcade Powered Source-Build infra re-clones the current repo somewhere in `artifacts/`, then runs msbuild on this source with source-build parameters.  This msbuild invocation is what I am referring to the "inner" build here.  As noted here this causes a conflict of the .globalconfig entries and therefore all settings are ignored.

Reply:
In CI, this seems unnecessary. Why not patch the current repo in-place❔

Even in a local build, dev could `git restore .` to undo the patches when they're done w/ source builds.

Reply:
Either way I think this problem will go away when we add our own CI and remove patches?

Reply:
This was a new source-build problem the ArPow effort caused.  The 5.0 source-build infra didn't encounter this because it doesn't have an inner clone - it clones the source in the source-build repo.  I could have introduced a patch to address the problem but the long term goal is to eliminate patches and since this was a is problem I wanted to find a permanent solution.  

One thing to call out here is the patches only get applied to the inner clone.  They get applied during each invocation of  source-build.  If a dev does 'git restore .' it isn't going to affect source-build because the patches aren't applied to the dev's repo.